starter-code/scripts/article.js

```diff
@@ -47,16 +47,22 @@ Article.loadAll = function(rawData) {
 // and process it, then hand off control to the View.
 Article.fetchAll = function() {
   if (localStorage.rawData) {
+    var lsdata = JSON.parse(localStorage.rawData)
     // When rawData is already in localStorage,
     // we can load it with the .loadAll function above,
     // and then render the index page (using the proper method on the articleView object).
-    Article.loadAll(); //TODO: What do we pass in to loadAll()?
+    Article.loadAll(lsdata); //TODO: What do we pass in to loadAll()?
     //TODO: What method do we call to render the index page?
+    articleView.initIndexPage();
   } else {
     // TODO: When we don't already have the rawData,
     // we need to retrieve the JSON file from the server with AJAX (which jQuery method is best for this?),
     // cache it in localStorage so we can skip the server call next time,
     // then load all the data into Article.all with the .loadAll function above,
```

Comment:
What is currently happening and why it is working is because without local storage, you go down to the else statement, get your JSON, then you put your data into local storage and call the same method again. Now you have local storage so your page is then getting the local storage, loading it and creating the index page. Instead of doing that, you should be loading and initing the index page inside that else statement.